### PR TITLE
cmake: Add a native build job on Ubuntu 20.04 + CMake 3.16

### PR DESCRIPTION
**Pull request by hebasto**
This PR:
1. Adds a CI job that uses the minimum required CMake 3.16 and tests the `ALLOW_HOST_PACKAGES` depends feature.
2. Fixes tests when Pyrhon is not available.

### CODE REVIEW

File: .github/workflows/cmake.yml [hunk 1]
```diff
@@ -79,6 +79,48 @@ jobs:
...
+    name: 'Ubuntu 20.04, CMake 3.16, Boost from depends'
+    runs-on: ubuntu-20.04
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+
+      - name: Remove preinstalled CMake
+        run: |
+          sudo rm /usr/local/bin/cmake /usr/local/bin/ctest
```
Comment (TheCharlatan) [inline .github/workflows/cmake.yml:92]: Is this something that comes with the github actions image? I do not have this package when I do:
```
docker pull ubuntu:20.04
docker run -ti --rm ubuntu:20.04 /bin/bash
```

**Review by TheCharlatan — COMMENTED** [review]
lgtm, just some questions.

File: .github/workflows/cmake.yml (hunk 1 above)
Comment (hebasto) [reply .github/workflows/cmake.yml:92]: > Is this something that comes with the github actions image?

It is.

File: cmake/tests.cmake [hunk 2]
```diff
@@ -1,18 +1,20 @@
-# Copyright (c) 2023 The Bitcoin Core developers
+# Copyright (c) 2023-present The Bitcoin Core developers
 # Distributed under the MIT software license, see the accompanying
-# file COPYING or http://www.opensource.org/licenses/mit-license.php.
+# file COPYING or https://opensource.org/license/mit/.
 
 include(CTest)
 
-if(TARGET bitcoin-util AND TARGET bitcoin-tx)
+if(TARGET bitcoin-util AND TARGET bitcoin-tx AND PYTHON_COMMAND)
```
[the remark this answers is not in the document]
Comment (hebasto) [reply cmake/tests.cmake:7]: Thanks! Added.

**Comment by hebasto** [comment]
@TheCharlatan

Thank you for your review! Your comments have been addressed.

**Review by TheCharlatan — APPROVED** [review]
ACK 53cb2c88b70de37ebab01fb22a6fdb16aa6caba7

**Pull request merged**